test/powershell/Modules/Microsoft.PowerShell.Utility/xml.tests.ps1

```diff
@@ -55,19 +55,19 @@
         }
 
         It "non filesystem path using <parameter>" -TestCases @(
-            @{parameter="literalPath"},
-            @{parameter="path"}
+            @{parameter="literalPath"; expectedError='ProcessingFile,Microsoft.PowerShell.Commands.SelectXmlCommand'},
+            @{parameter="path"; expectedError='PathNotFound,Microsoft.PowerShell.Commands.SelectXmlCommand'}
         ) {
-            param($parameter)
-            $__data = "abcdefg"
-            $params = @{$parameter="variable:__data"}
-            { Select-XML @params "Root" | Should BeErrorId 'ProcessingFile,Microsoft.PowerShell.Commands.SelectXmlCommand' }
+            param($parameter, $expectedError)
+            $file = Join-Path -Path $testDrive -ChildPath 'abcdefg'
+            $params = @{$parameter=$file}
+            { Select-XML @params "Root" -ErrorAction Stop} | ShouldBeErrorId $expectedError
         }
 
         It "Invalid xml file" {
             $testfile = "$testdrive/test.xml"
             Set-Content -Path $testfile -Value "<a><b>"
-            { Select-Xml -Path $testfile -XPath foo | ShouldBeErrorId 'ProcessingFile,Microsoft.PowerShell.Commands.SelectXmlCommand' }
+            { Select-Xml -Path $testfile -XPath foo -ErrorAction Stop} | ShouldBeErrorId 'ProcessingFile,Microsoft.PowerShell.Commands.SelectXmlCommand'
```

Comment:
Do we have a bug in ShouldBeErrorId if this test was passing without `-ErrorAction Stop`?

Reply:
I believe the test can only false pass if  Select-Xml return AutomationNull.

Reply:
This test was passing because it wasn't actually doing anything before. The closing } was at the end of the line resulting in a script block being created versus a script block being passed to ShouldBeErrorId.  Both tests fixed in this file had the same problem.
As far as -ErrorAction Stop; Select-Xml is not a terminating error so ShouldBeErrorId doesn't catch it.

Reply:
Fixed.  Using -ErrorVariable instead of -ErrorAction Stop.